paddle/fluid/framework/selected_rows.cc

```diff
@@ -69,5 +116,66 @@ void DeserializeFromStream(std::istream& is, SelectedRows* selected_rows,
   TensorFromStream(is, selected_rows->mutable_value(), dev_ctx);
 }
 
+bool SelectedRows::HasKey(int64_t key) const {
+  return std::find(rows_.begin(), rows_.end(), key) == rows_.end() ? false
+                                                                   : true;
+}
+
+bool SelectedRows::Get(int64_t key, framework::Tensor* value,
+                       int64_t row) const {
```

Comment:
row is the same as key?

Reply:
No, it's the offset of the given value pointer, maybe we can rename it to `offset`...

Reply:
Done.